docs/configuration.md

```diff
@@ -629,3 +629,104 @@ window.$docsify = {
   topMargin: 90, // default: 0
 };
 ```
+
+## vueComponents
+
+- type: `Object`
+
+Registers Vue components using the component name as the key with an object containing Vue options as the value.
+
+```js
+window.$docsify = {
+  vueComponents: {
+    'button-counter': {
+      template: `
+        <button @click="count += 1">
+          You clicked me {{ count }} times
+        </button>
+      `,
+      data() {
+        return {
+          count: 0,
+        };
+      },
+    },
+  },
+};
+```
```

Comment:
Why would we want to use `vueMountOptions` with a selector, when we could specify components using `vueComponents`?

Trying to get a more clear picture.

Reply:
Fair question.

Here are the first few that come to mind:

1. Vue supports mounting both DOM elements and components. So should we. All of the reasons a Vue user might opt to mount a DOM element instead of creating a Vue component apply here.

1. Mounting DOM elements is the first concept found in the [Vue docs](https://vuejs.org/v2/guide/) and introductory Vue tutorials. It is the fastest and simplest way to be productive with a library like Vue and requires the least amount of expertise. Allowing code from docs and tutorials to be easily ported to docsify is an important consideration for our users--especially junior and non-developer users.

1. Components are typically presented as reusable building blocks that combine to make larger applications. Obviously this doesn't have to be the case, but it's how they are often conceptualized. Forcing users to think in "component" terms can be confusing when they are just trying to do something simple task like render data on a docsify page.

1. There are technical and UX differences between mounted DOM elements and components. For example, `vueMountOptions` allows mounting content found directly in your markdown file:

   ```markdown
   # Title

   <p id="my-content">Here is some Vue content: {{ message }}</p>
   ```
   ```js
   vueMountOptions: {
     '#my-content': {
       data() {
         return {
           message: 'Hello, World!'
         };
       }
     }
   }
   ```

   Vue components can do this as well, but doing so requires adding an `inline-template` attribute to the component tag which is not intuitive. 

   ```markdown
   # Title

   <my-component inline-template>
     <p>Here is some Vue content: {{ message }}</p>
   </my-component>
   ```

   More typically, components specify their markup using the `template` property:

   ```markdown
   # Title

   <my-component />
   ```
   ```js
   vueComponents: {
     'my-component': {
       template: `<p>Here is some Vue content: {{ message }}</p>`,
       data() {
         return {
           count: 0,
         };
       }
     }
   }
   ```

   This is not a big deal when building Vue apps outside of docsify, but with docsify it means `template` markup cannot contain markdown, [will not be processed by docsify plugins](https://github.com/jhildenbiddle/docsify-tabs/issues/31), will not have syntax highlighting or linting while authoring, will not render default content outside of docsify (e.g., GitHub), etc. You can work around some of these things, but doing requires a level of expertise that not all docsify users will have.

As for our and why you would choose one over the other, my intention was keep it light on our end and refer people to the Vue docs for more information. If you think we need to clarify why users should choose one over the other when using docsify, I can work on that.

Reply:
Thanks for that explanation! I know about the starting the root of a Vue tree and components used in the Vue tree, but these are two features additional to Vue (for Docsify).

Knowing Vue, I know I can put `new Vue` in a script that I can load with `<script src="...">` on every page.

I suppose what I'm really asking is the following.

Here are two ways to basically do the same thing with the new system. 

**Method 1:**

```html
<div id="my-content">

## {{ title }}
Here is some Vue content: {{ message }}

</div>
```

```js
vueMounts: {
  '#my-content': {
    data: () => ({
      title: 'My Title',
      message: 'Hello, World!'
    })
  }
}
```

**Method 2:**

```html
<my-content inline-template>

## {{ title }}
Here is some Vue content: {{ message }}

</my-content>
```

```js
vueComponents: {
  'my-content': {
    data: () => ({
      title: 'My Title',
      message: 'Hello, World!'
    })
  }
}
```

Both have about the same ease-of-use (learning to use `inline-template` is as easy as writing an id or class name). The question is, why have both?

Is the difference only the inline-template vs no inline-template thing? If so, is it worth having two options just for that small difference?